pkg/db/db.go

```diff
@@ -346,3 +346,38 @@ func ReadCorpus(filename string, target *prog.Target) (progs []*prog.Prog, err e
 	}
 	return progs, nil
 }
+
+type DeserializeFailure struct {
+	File string
+	Err  error
+}
+
+func Merge(into string, other []string, target *prog.Target) ([]DeserializeFailure, error) {
+	dstDB, err := Open(into, false)
+	if err != nil {
+		return nil, fmt.Errorf("failed to open database: %w", err)
+	}
+	var failed []DeserializeFailure
+	for _, add := range other {
+		if addDB, err := Open(add, false); err == nil {
+			for key, rec := range addDB.Records {
+				dstDB.Save(key, rec.Val, rec.Seq)
+			}
+			continue
+		} else if target == nil {
```

Comment:
While Open(add, false) returns no error, you "continue".
In case of any error you want to check for `if target == nil`.

I think you can write "if target == nil" instead of "else it target == nil".

Reply:
That's WAI (exactly as in the original code). If it's a DB, it's opened as DB and then merged, otherwise it's treated as a seed file.

Reply:
Ah, now I see what you meant. I've split the if-else.